Assets/Scripts/Checkpoint.cs

```diff
@@ -3,22 +3,34 @@
 using UnityEngine;
 using UnityEngine.Animations;
 
-public class Checkpoint : MonoBehaviour {
-    [SerializeField]
-    public bool isActivated = false;
-   
-    public Animator anim;
-
-    //creates a variable that refers to the PlayerController
-    PlayerController player;
+public class Checkpoint : MonoBehaviour
+{    
+    private bool isActivated;   
+    private Animator anim;
+    private PlayerController player;
+    /// <summary>
+    /// Property Used to Activate or Deactivate checkpoints
+    /// </summary>
+    public bool IsActivated
+    {
+        get
+        {
+            return isActivated;
+        }
+        set
+        {
+            isActivated = value;
+        }
+    }
 
     private void Start()
     {
-        anim = this.GetComponent<Animator>();       
+        IsActivated = false;
+        anim = GetComponent<Animator>();
     }
 
     private void Update()
-    {
+    {        
         UpdateAnimation();
```

Comment:
Do you need to call this every frame? If it only needs to be called when IsActivated is set, you can call it from IsActivated's setter.